response_operations_ui/views/accounts.py

```diff
@@ -104,66 +101,97 @@ def change_account_name():
 @account_bp.route("/change-email", methods=["GET", "POST"])
 @login_required
 def change_email():
-    url_safe_serializer = URLSafeSerializer(app.config["SECRET_KEY"])
     form = ChangeEmailForm()
-    user_id = session["user_id"]
-    user_from_uaa = uaa_controller.get_user_by_id(user_id)
+
     if request.method == "POST" and form.validate():
-        if form.data["email_address"] != user_from_uaa["emails"][0]["value"]:
-            personalisation = {
-                "first_name": user_from_uaa["name"]["givenName"],
-                "value_name": "email",
-                "changed_value": form.data["email_address"],
-            }
-            try:
-                logger.info("Sending verification email", email=obfuscate_email(form.data["email_address"]))
-                token_dict = {"email": form.data["email_address"], "user_id": user_id}
-                send_update_account_email(token_dict, user_from_uaa["name"]["givenName"])
-                logger.info("Sending notification email")
-                NotifyController().request_to_notify(
-                    email=user_from_uaa["emails"][0]["value"],
-                    template_name="update_account_details",
-                    personalisation=personalisation,
-                )
-                logger.info(
-                    "Successfully sent email change email",
-                    encoded_email=url_safe_serializer.dumps(form.data["email_address"]),
-                )
+        user_id = session["user_id"]
+        uaa_user = uaa_controller.get_user_by_id(user_id)
+
+        if uaa_user is None:
+            return render_template("request-new-account-error.html")
+
+        current_email = uaa_user["emails"][0]["value"]
+        new_email = form.data["email_address"]
+
+        if current_email != new_email and _can_change_username_and_email(new_email):
+            first_name = uaa_user["name"]["givenName"]
+
+            if _notify_account_update_emails_sent(current_email, new_email, first_name, user_id):
                 flash(
-                    "A verification email has been sent. You will need to login to change your email",
+                    "A verification email has been sent.",
                     category="successful_signout",
                 )
                 return redirect(url_for("logout_bp.logout"))
-            except NotifyError as e:
-                logger.error("Error sending 'email change' email to Notify Gateway", msg=e.description)
-                flash("Something went wrong while updating your email. Please try again", category="error")
+
+            flash("Something went wrong while updating your email. Please try again", category="error")
         else:
-            return redirect(url_for("account_bp.get_my_account"))
+            logger.info(f"Unable to change username and email for {user_id}, email already used")
```

Comment:
Feasibly you could use the new email here, but because I would have to obfuscate I didn't feel it added much